src/Mvc/src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Binders/CollectionModelBinder.cs

```diff
@@ -319,8 +358,12 @@ private Task<CollectionResult> BindComplexCollection(ModelBindingContext binding
             else
             {
                 indexNamesIsFinite = false;
-                indexNames = Enumerable.Range(0, int.MaxValue)
-                                       .Select(i => i.ToString(CultureInfo.InvariantCulture));
+                var limit = _maxModelBindingCollectionSize == int.MaxValue ?
+                    int.MaxValue :
```

Comment:
suggestion - use `int?` to make the sentinel value clear

Reply:
This isn't a sentinel. True that people can set insanely large limits and turn things off effectively. But, &hellip;
- ` _maxModelBindingCollectionSize + 1` would overflow if this wasn't a ternary expression.
- always need a value for the `Range(...)` expression just below